src/manybody_allegro.F

```diff
@@ -0,0 +1,480 @@
+      t_edge_index(:, :) = TRANSPOSE(temp_edge_index)
+      DEALLOCATE (temp_edge_index, edge_index)
+
+      lattice = cell%hmat/pot%set(1)%allegro%unit_cell_val
+
+      iat_use = 0
+      ALLOCATE (pos(3, n_atoms_use), atom_types(n_atoms_use))
+
+      DO iat = 1, n_atoms_use
+         IF (.NOT. use_atom(iat)) CYCLE
+         iat_use = iat_use + 1
+         atom_types(iat_use) = particle_set(iat)%atomic_kind%kind_number - 1
+         pos(:, iat) = REAL(r_last_update_pbc(iat)%r(:)/pot%set(1)%allegro%unit_coords_val, kind=sp)
+      END DO
+
+      CALL torch_dict_create(inputs)
+      CALL torch_dict_insert(inputs, "pos", pos)
+      CALL torch_dict_insert(inputs, "edge_index", t_edge_index)
+      CALL torch_dict_insert(inputs, "edge_cell_shift", new_edge_cell_shifts)
+      CALL torch_dict_insert(inputs, "cell", REAL(lattice, kind=sp))
+      CALL torch_dict_insert(inputs, "atom_types", atom_types)
+
+      CALL torch_dict_create(outputs)
+
+      CALL torch_model_eval(allegro_data%model, inputs, outputs)
+
+      CALL torch_dict_get(outputs, "atomic_energy", atomic_energy)
+      CALL torch_dict_get(outputs, "forces", forces)
+
+      allegro_data%force(:, :) = REAL(forces(:, :), kind=dp)*allegro%unit_forces_val
+
+      pot_allegro = 0.0_dp
+
+      DO iat_use = 1, SIZE(unique_list_a)
+         i = unique_list_a(iat_use)
+         pot_allegro = pot_allegro + REAL(atomic_energy(1, i), kind=dp)*allegro%unit_energy_val
+      END DO
+
+      CALL torch_dict_release(inputs)
+      CALL torch_dict_release(outputs)
+
+      DEALLOCATE (pos, t_edge_index, edge_cell_shifts, atom_types, atomic_energy)
+      DEALLOCATE (new_edge_cell_shifts)
+      DEALLOCATE (forces)
+
+      CALL timestop(handle)
+   END SUBROUTINE allegro_energy_store_force_virial
+
+! **************************************************************************************************
+!> \brief ...
+!> \param fist_nonbond_env ...
+!> \param f_nonbond ...
+!> \param pv_nonbond ...
+!> \param use_virial ...
+! **************************************************************************************************
+   SUBROUTINE allegro_add_force_virial(fist_nonbond_env, f_nonbond, pv_nonbond, use_virial)
+
+      TYPE(fist_nonbond_env_type), POINTER               :: fist_nonbond_env
+      REAL(KIND=dp), DIMENSION(:, :), INTENT(INOUT)      :: f_nonbond, pv_nonbond
+      LOGICAL, INTENT(IN)                                :: use_virial
+
+      INTEGER                                            :: iat, iat_use
+      REAL(KIND=dp), DIMENSION(3, 3)                     :: virial
+      TYPE(allegro_data_type), POINTER                   :: allegro_data
+
+      CALL fist_nonbond_env_get(fist_nonbond_env, allegro_data=allegro_data)
+
+      IF (use_virial) THEN
+         virial(1, 1) = 0.0_dp; virial(1, 2) = 0.0_dp; virial(1, 3) = 0.0_dp
+         virial(2, 1) = 0.0_dp; virial(2, 2) = 0.0_dp; virial(2, 3) = 0.0_dp
+         virial(3, 1) = 0.0_dp; virial(3, 2) = 0.0_dp; virial(3, 3) = 0.0_dp
```

Comment:
I am aware that this branch is not yet properly implemented. But just write `virial = 0.0_dp` and below `pv_nonbond = pv_nonbond + virial`

Reply:
Done, thanks.